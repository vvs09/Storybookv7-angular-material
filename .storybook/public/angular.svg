<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
