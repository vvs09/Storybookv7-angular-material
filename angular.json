{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}