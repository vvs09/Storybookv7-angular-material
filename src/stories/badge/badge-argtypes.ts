export const badgeArgTypes = {
  color: {
    control: 'radio',
    options: ['accent', 'primary', 'warn'],
    description: 'Theme color palette for the component.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'null' },
      type: {
        summary: 'string',
      },
    },
  },
  disabled: {
    control: 'boolean',
    description: 'Whether the badge is disabled.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  content: {
    control: 'text',
    description: 'Whether ripples are disabled.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'null' },
      type: {
        summary: 'string | number | undefined | null',
      },
    },
  },
  description: {
    control: 'text',
    description:
      'Message used to describe the decorated element via aria-describedby',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  position: {
    control: 'select',
    options: [
      'above after',
      'above before',
      ' below before',
      'below after ',
      'before',
      'after',
      'above',
      'below',
    ],
    description:
      'Position the badge should reside. Accepts any combination of above | below and before|after.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '' },
      type: {
        summary:
          'above after| above before | below before| below after | before | after | above | below',
      },
    },
  },
  size: {
    control: 'radio',
    options: ['small', 'medium', 'large'],
    description: 'Size of the badge. Can be small, medium, or large.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '' },
      type: {
        summary: "'small' | 'medium' | 'large'",
      },
    },
  },
  hidden: {
    control: 'boolean',
    description: 'Whether the badge is hidden.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  overlap: {
    control: 'boolean',
    description: 'Whether the badge should overlap its contents or not',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
};
