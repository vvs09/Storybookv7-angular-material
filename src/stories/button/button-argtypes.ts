export const buttonArgTypes= {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'null' },
        type: {
          summary: 'string',
        },
      },
    },
    disabled: {
      control: 'boolean',
      description:
        'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    buttonType: {
      control: 'select',
      options: [
        'mat-button',
        'mat-raised-button',
        'mat-stroked-button',
        'mat-flat-button',
      ],
      description: 'Type of buton.',
      table: {
        defaultValue: { summary: 'mat-button' },
        type: {
          summary: 'string',
        },
      },
    },
    label: {
      control: 'text',
      description: 'Button label.',
      table: {
        defaultValue: { summary: 'button' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaDisabled: {
      control: 'boolean',
      description: 'aria-disabled value of the button.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'undefined' },
        type: {
          summary: 'boolean | undefined',
        },
      },
    },
    disabledInteractive: {
      control: 'boolean',
      description:
        'Natively disabled buttons prevent focus and any pointer events from reaching the button. In some scenarios this might not be desirable, because it can prevent users from finding out why the button is disabled (e.g. via tooltip). Enabling this input will change the button so that it is styled to be disabled and will be marked as aria-disabled, but it will allow the button to receive events and focus. Note that by enabling this, you need to set the tabindex yourself if the button isnt meant to be tabbable and you have to prevent the button action (e.g. form submissions).',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
  }