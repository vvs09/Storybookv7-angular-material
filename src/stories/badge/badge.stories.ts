import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatBadge, MatBadgeModule } from '@angular/material/badge';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { badgeArgTypes } from './badge-argtypes';

const meta: Meta<MatBadge & { label: string }> = {
  title: 'Components/Badge',
  component: MatBadge,
  tags: ['autodocs'],
  argTypes:badgeArgTypes,
  decorators: [
    moduleMetadata({
      imports: [MatBadgeModule, MatButtonModule, MatIconModule],
    }),
  ],
  render:({label,...args}) => ({
    props: args,
    template: `
    <div matBadge style="display:inline" ${argsToTemplate(args)}>${label}</div>
   `,
  }),
};

export default meta;
type Story = StoryObj<MatBadge & { label: string }>;

export const accent: Story = {
  args: {
    label: 'Text with a badge',
    overlap: false,
    position: 'above after',
    content: '8',
    color: 'accent',
    disabled: false,
    description: 'badge',
    size: 'small',
    hidden: false,
  },
};

export const Primary: Story = {
  args: {
    ...accent.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...accent.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...accent.args,
    disabled: true,
  },
};

export const overlap: Story = {
  args: {
    ...accent.args,
    overlap: true,
  },
};

export const content: Story = {
  args: {
    ...accent.args,
    content: '67',
  },
};

export const Position: Story = {
  args: {
    ...accent.args,
    position: 'before',
  },
};

export const Size: Story = {
  args: {
    ...accent.args,
    size: 'large',
  },
};

export const hidden: Story = {
  args: {
    ...accent.args,
    hidden: true,
  },
};
