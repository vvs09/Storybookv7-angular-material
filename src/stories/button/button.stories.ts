import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatButtonModule, MatButton } from '@angular/material/button';
import { buttonArgTypes } from './button-argtypes';

const meta: Meta<MatButton & { label: string; buttonType: string }> = {
  title: 'Components/Button',
  component: MatButton,
  tags: ['autodocs'],
  argTypes: {
    ...buttonArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [MatButtonModule],
    }),
  ],
  render: (args) => ({
    props: args,
    template: `
    <button  *ngIf="buttonType === 'mat-button'" mat-button  ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-raised-button' " mat-raised-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-stroked-button' " mat-stroked-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-flat-button' "  mat-flat-button ${argsToTemplate(
      args
    )} >{{label}}</button>
\    `,
  }),
};

export default meta;
type Story = StoryObj<MatButton & { label: string; buttonType: string }>;

export const accent: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDisabled: false,
    disabledInteractive: false,
    disableRipple: false,
    label: 'Button',
    buttonType: 'mat-button',
  },
};

export const Primary: Story = {
  args: {
    ...accent.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...accent.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...accent.args,
    disabled: true,
  },
};

export const matRaisedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-raised-button',
  },
};

export const matRaisedButtonPrimary: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'primary',
  },
};

export const matRaisedButtonWarn: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'warn',
  },
};

export const MatRaisedButtonDisabled: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    disabled: true,
  },
};

export const MatStrokedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-stroked-button',
  },
};

export const MatStrokedButtonPrimary: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'primary',
  },
};

export const MatStrokedButtonWarn: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'warn',
  },
};

export const MatStrokedButtonDisabled: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    disabled: true,
  },
};

export const MatFlatButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-flat-button',
  },
};

export const MatFlatButtonPrimary: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'primary',
  },
};

export const MatFlatButtonWarn: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'warn',
  },
};

export const MatFlatButtonDisabled: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    disabled: true,
  },
};
