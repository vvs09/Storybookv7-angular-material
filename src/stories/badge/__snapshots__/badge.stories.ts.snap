// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Badge Disabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-accent mat-badge-small mat-badge-disabled"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-3"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge Position smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-accent mat-badge-before mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-6"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge Primary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-primary mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-1"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge Size smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-accent mat-badge-large"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-7"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge Warn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-warn mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-2"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge accent smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-accent mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-0"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge content smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-accent mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-5"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      67
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge hidden smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-above mat-badge-after mat-badge-hidden mat-badge-accent mat-badge-small"
       style="display: inline;"
       ng-reflect-content
       hidden
  >
    Text with a badge
    <span id="mat-badge-content-8"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;

exports[`Components/Badge overlap smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <div matbadge
       class="mat-badge mat-badge-overlap mat-badge-above mat-badge-after mat-badge-accent mat-badge-small"
       style="display: inline;"
       ng-reflect-content
  >
    Text with a badge
    <span id="mat-badge-content-4"
          aria-hidden="true"
          class="mat-badge-content mat-badge-active"
    >
      8
      <span class="cdk-visually-hidden">
        badge
      </span>
    </span>
  </div>
</storybook-root>
`;
